lib/zstd.h

```diff
@@ -1740,17 +1740,22 @@ ZSTDLIB_STATIC_API size_t ZSTD_checkCParams(ZSTD_compressionParameters params);
  *  This function never fails (wide contract) */
 ZSTDLIB_STATIC_API ZSTD_compressionParameters ZSTD_adjustCParams(ZSTD_compressionParameters cPar, unsigned long long srcSize, size_t dictSize);
 
+/*! ZSTD_CCtx_setCParams() :
+ *  Set all parameters provided within @cparams into the working @cctx
+ * @return 0 on success, or an error code (can be checked with ZSTD_isError()) */
```

Comment:
It would be good to document that the user will lose out on some cParam validation: https://github.com/facebook/zstd/blob/89342d1e076e86e59d143ce2fa3b8e8af6f65b03/lib/compress/zstd_compress.c#L670-L944